### PR TITLE
Make sure that check_struct is called where needed

### PR DESCRIPTION
**Pull request by almarklein**

### CODE REVIEW

File: codegen/apipatcher.py [hunk 1]
```diff
@@ -42,7 +42,11 @@ def patch_backend_api(code):
         base_api_code = f.read().decode()
 
     # Patch!
-    for patcher in [CommentRemover(), BackendApiPatcher(base_api_code)]:
+    for patcher in [
+        CommentRemover(),
+        BackendApiPatcher(base_api_code),
+        BackendApiStructValidaitonChecker(),
```
Comment (Korijn) [inline codegen/apipatcher.py:48]: Typo :)

Comment (almarklein) [reply codegen/apipatcher.py:48]: how did you even spot that?

File: wgpu/backends/rs.py [hunk 2]
```diff
@@ -862,7 +880,8 @@ def create_render_pipeline(
             assert (
                 depth_stencil.get("format", None) is not None
             ), "depth_stencil needs format"
-            stencil_front = depth_stencil.get("front", {})
+            stencil_front = depth_stencil.get("stencil_front", {})
```
Comment (almarklein) [inline wgpu/backends/rs.py:883]: I changed this from stencil_front -> front earlier, I don't know why I thought this was changed.

**Comment by almarklein** [comment]
Ready! Already caught a few bugs in our own code!

**Pull request merged**